broke-client/src/main/java/org/dsngroup/broke/client/BlockClient.java

```diff
@@ -39,16 +39,18 @@
     private OutputStream clientOutputStream;
 
     /**
-     * @param
+     * @param topic topic to publish
+     * @param qos qos of transmission
+     * @param criticalOption TODO: critical option mechanism
+     * @param payload payload of the message
      */
     public void publish(String topic, int qos, int criticalOption, String payload) throws Exception {
-        PublishMessage msg = (PublishMessage) MessageBuilder
-                .build("PUBLISH\r\nQoS:"+qos+",Critical-Option:"+criticalOption+",Topic:hello\r\n"+payload+"\r\n");
         // TODO: Send this message
-        clientOutputStream.write(("PUBLISH\r\nQoS:"+qos+",Critical-Option:"+criticalOption+",Topic:hello\r\n"+payload+"\r\n")
+        clientOutputStream.write(("PUBLISH\r\nQoS:"+qos+",Topic:"+topic+"\r\n"+payload+"\r\n")
                 .getBytes(Charset.forName("UTF-8")));
```

Comment:
In the future, I think it's better wrap this ugly message concat encapsulated into message builder, it can also use `StringBuilder` to avoid redundant temporary string instances.